### PR TITLE
[LIBSAMPLERATE] Updated to latest version

### PR DESCRIPTION
**Pull request by BenNottelling**
## Purpose
I noticed that our branch of libsamplerate was outdated, so I figured I'd merge in the latest version, mostly so I can understand a little more on the process

ReactOS builds and runs fine with these changes
JIRA issue: none

### CODE REVIEW

File: sdk/lib/3rdparty/libsamplerate/CMakeLists.txt [hunk 1]
```diff
@@ -9,3 +9,4 @@ list(APPEND SOURCE
 add_library(libsamplerate ${SOURCE})
 add_dependencies(libsamplerate psdk)
 add_pch(libsamplerate precomp.h SOURCE)
+set(CMAKE_C_FLAGS "-std=c99")
```
Comment (HBelusca) [inline sdk/lib/3rdparty/libsamplerate/CMakeLists.txt]: That won't do it with MSVC.

File: sdk/lib/3rdparty/libsamplerate/samplerate.c [hunk 2]
```diff
@@ -6,7 +6,15 @@
...
-#include "precomp.h"
+#include	<stdio.h>
+#include	<stdlib.h>
+#include	<string.h>
+
+#include	"config.h"
+
+#include	"samplerate.h"
+#include	"float_cast.h"
+#include	"common.h"
```
Comment (HBelusca) [inline sdk/lib/3rdparty/libsamplerate/samplerate.c]: Isn't it possible to use again the precompiled header? (if needed, to add in it the missing new headers)

File: sdk/lib/3rdparty/libsamplerate/src_linear.c [hunk 3]
```diff
@@ -6,10 +6,17 @@
...
 */
 
-#include "precomp.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "config.h"
+#include "float_cast.h"
+#include "common.h"
```
Comment (HBelusca) [inline sdk/lib/3rdparty/libsamplerate/src_linear.c]: Isn't it possible to use again the precompiled header? (if needed, to add in it the missing new headers)

File: sdk/lib/3rdparty/libsamplerate/src_sinc.c [hunk 4]
```diff
@@ -6,7 +6,13 @@
...
 */
 
-#include "precomp.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "config.h"
+#include "float_cast.h"
+#include "common.h"
```
Comment (HBelusca) [inline sdk/lib/3rdparty/libsamplerate/src_sinc.c]: Isn't it possible to use again the precompiled header? (if needed, to add in it the missing new headers)

File: sdk/lib/3rdparty/libsamplerate/src_zoh.c [hunk 5]
```diff
@@ -6,10 +6,17 @@
...
 */
 
-#include "precomp.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "config.h"
+#include "float_cast.h"
+#include "common.h"
```
Comment (HBelusca) [inline sdk/lib/3rdparty/libsamplerate/src_zoh.c]: Isn't it possible to use again the precompiled header? (if needed, to add in it the missing new headers)

File: sdk/lib/3rdparty/libsamplerate/src_zoh.c [hunk 6]
```diff
@@ -192,3 +200,21 @@ zoh_reset (SRC_PRIVATE *psrc)
 	return ;
 } /* zoh_reset */
 
+static int
+zoh_copy (SRC_PRIVATE *from, SRC_PRIVATE *to)
+{
+	if (from->private_data == NULL)
+		return SRC_ERR_NO_PRIVATE ;
+
+	ZOH_DATA *to_priv = NULL ;
```
Comment (HBelusca) [inline sdk/lib/3rdparty/libsamplerate/src_zoh.c]: That will fail with MSVC (variable declaration after code).

File: sdk/lib/3rdparty/libsamplerate/samplerate.c (hunk 2 above)
Comment (BenNottelling) [reply sdk/lib/3rdparty/libsamplerate/samplerate.c]: Yup, I was just keeping the files as original from source :)

**Comment by ThFabba** [comment]
Thank you. However please note that verifying the correctness of a sync like this is actually more work than performing the sync, which means it isn't particularly useful unless someone with merge rights does it.

**Comment by BenNottelling** [comment]
> Thank you. However please note that verifying the correctness of a sync like this is actually more work than performing the sync, which means it isn't particularly useful unless someone with merge rights does it.

Why is it more work? Just trying to understand here

**Comment by ThFabba** [comment]
To verify that you (or anyone else) did the sync correctly and did not add or omit code (in larger syncs, even added a backdoor), I have to do the sync myself and then compare. That is more work than just doing it myself, so a PR of this type is not useful to the project. Sorry.

**Pull request closed without merging**